diff --git a/src/Makefile.am b/src/Makefile.am
index b7d6409..1fb55d6 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -626,7 +626,7 @@ protoc_outputs =                                                  \
   google/protobuf/util/json_format_proto3.pb.cc                   \
   google/protobuf/util/json_format_proto3.pb.h
 
-BUILT_SOURCES = $(protoc_outputs)
+#BUILT_SOURCES = $(protoc_outputs)
 
 if USE_EXTERNAL_PROTOC
 
